tests/BitcoinGold/TWSignerTests.cpp

```diff
@@ -0,0 +1,84 @@
+// Copyright © 2017-2020 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include <TrustWalletCore/TWAnyAddress.h>
+#include <TrustWalletCore/TWAnySigner.h>
+#include <TrustWalletCore/TWBitcoinSigHashType.h>
+#include <gtest/gtest.h>
+
+#include "BitcoinGold/SegwitAddress.h"
+#include "proto/Bitcoin.pb.h"
+#include "Bitcoin/OutPoint.h"
+#include "Bitcoin/Script.h"
+#include "Bitcoin/Transaction.h"
+#include "Bitcoin/TransactionBuilder.h"
+#include "Bitcoin/TransactionSigner.h"
+#include "HexCoding.h"
+#include "../interface/TWTestUtilities.h"
+
+using namespace TW;
+using namespace TW::Bitcoin;
+
+TEST(TWBitcoinGoldSigner, SignTransaction) {
+    const int64_t amount = 10000;
+
+    // Setup input
+    Proto::SigningInput input;
+    input.set_coin_type(TWCoinTypeBitcoinGold);
+    input.set_hash_type(TWBitcoinSigHashTypeFork | TWBitcoinSigHashTypeAll | (79 << 8));
```

Comment:
When signing a BTG transaction, make sure to bitwise or (79 << 8)  to the hash_type to enable the proper fork_id.